pkg/tektonlog/tektonlog.go

```diff
@@ -304,6 +292,57 @@ func (t *TektonLogger) getRunningBuildLogs(ctx context.Context, pa *v1.PipelineA
 	return nil
 }
 
+func (t *TektonLogger) collectStages(ctx context.Context, pipelineRuns []*tektonapis.PipelineRun) []stageTime {
+	var podTimes []stageTime
+	for _, prInitial := range pipelineRuns {
+		//we need fresh pipeline to be able consume newly executed tasks/pods
+		pr, _ := t.TektonClient.TektonV1beta1().PipelineRuns(t.Namespace).Get(ctx, prInitial.Name, metav1.GetOptions{})
+		for _, taskStatus := range pr.Status.PipelineSpec.Tasks {
```

Comment:
is the `Status.PipelineSpec` field always set? In my v2 cluster it's not - maybe it is only set in v3? I was thinking about retrieving the Pipeline CRD to retrieve the list of tasks

Reply:
or maybe we can do it as a fallback: if the `Status.PipelineSpec` field is empty, then load the Pipeline CRD

Reply:
`Status.PipelineSpec` is always set in my case and contains list of all tasks inside pipeline

Reply:
ok I'll add the fallback to support jxv2 in a different PR later